web/packages/teleport/src/SessionRecordings/view/player/tty/AspectFitRatio.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * Teleport
+ * Copyright (C) 2025 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import type { ITerminalAddon, Terminal } from '@xterm/xterm';
+import type { IRenderDimensions } from '@xterm/xterm/src/browser/renderer/shared/Types';
+
+/**
+ * AspectFitAddon is a xterm.js addon that resizes the terminal to fit within its parent element
+ * while maintaining the specified aspect ratio defined by cols and rows.
+ * It uses the same approach as xterm's fit addon (accessing the _renderService and its dimensions).
+ * It applies CSS transforms to scale and center the terminal within its parent element.
+ */
+export class AspectFitAddon implements ITerminalAddon {
+  private _terminal: Terminal | undefined;
+
+  public activate(terminal: Terminal): void {
+    this._terminal = terminal;
+  }
+
+  public dispose(): void {}
+
+  public fitWithAspectRatio(cols: number, rows: number): void {
+    if (
+      !this._terminal ||
+      !this._terminal.element ||
+      !this._terminal.element.parentElement
+    ) {
+      return;
+    }
+
+    const core = (this._terminal as any)._core;
+    const dims: IRenderDimensions = core._renderService.dimensions;
+
+    if (dims.css.cell.width === 0 || dims.css.cell.height === 0) {
+      return;
+    }
+
+    const parentElementStyle = window.getComputedStyle(
+      this._terminal.element.parentElement
+    );
+    const parentElementHeight = parseInt(
+      parentElementStyle.getPropertyValue('height')
+    );
+    const parentElementWidth = Math.max(
+      0,
+      parseInt(parentElementStyle.getPropertyValue('width'))
+    );
```

Comment:
Why is width and height computed differently here?

Reply:
Not sure. Most of this is copied from xterm's fit addon, and I didn't want to mess with it as it worked